libs/cart/guides/state.md

```diff
@@ -0,0 +1,86 @@
+# State
```

Comment:
I feel like all of this could go in a general daffodil doc, because all packages will need it

Reply:
I think we should have a general doc that talks about Daffodil state architecture but this guide is talking about cart-specific stuff. This guide will be somewhat redundant but I don't think thats bad.